utilsplot.py

```diff
@@ -199,19 +194,29 @@ def bode(G, w_start=-2, w_end=2, axlim=None, points=1000, margin=False):
     plt.subplot(2, 1, 2)
     phaseangle = utils.phase(G(s), deg=True)
     plt.semilogx(w, phaseangle)
-    if margin:
-        plt.axvline(wc, color='black')
-        plt.text(wc,
-                 numpy.average([numpy.max(phaseangle), numpy.min(phaseangle)]),
-                 '|G(jw)| = 1')
     plt.axhline(-180., color='red', linestyle='--')
     plt.axis(axlim)
     plt.grid()
     plt.ylabel('Phase')
     plt.xlabel('Frequency [rad/unit time]')
 
-    return GM, PM
+    # Phase and Gain Margins
+    if margin:
+        for subplot_num, freq, bounds, legend in zip(
```

Comment:
I feel like it is more natural here construct the lines like this:

```
lines = ((1, wc, phaseangle, r'$\angle$G(jw) = -180$\degree$'), 
         (2, w_180, gains, '|G(jw)| = 1'))
for subplot_num, freq, bounds, legend in lines:
```

Reply:
That is better, I'll remember the pattern. I'm used to using zip() for long lists and I defaulted to it.